modules/apm/NAMING.md

```diff
@@ -1,82 +1,115 @@
+### Hierarchy and segments
 
-Example:
-* prefer `es.indices.docs.deleted.total `to `es.indices.total.deleted.docs`
-* This way you can later add` es.indices.docs.total, es.indices.docs.ingested.total`, etc.)
+Always use `es.` as the root segment to easily discover ES metrics and avoid the possibility of name clashes.
+Follow with a module name, team or area of code, e.g. `snapshot, repositories, indices, threadpool` using existing terminology (whether singular and plural).
 
-Prefix metrics:
-* Always use `es` as our root application name: this will give us a separate namespace and avoid any possibility of clashes with other metrics, and quick identification of Elasticsearch metrics on a dashboard.
-* Follow the root prefix with a simple module name, team or area of code. E.g. `snapshot, repositories, indices, threadpool`. Notice the mix of singular and plural - here this is intentional, to reflect closely the existing names in the codebase (e.g. `reindex` and `indices`)
-* In building a metric name, look for existing prefixes (e.g. module name and/or area of code, e.g. `blob_cache`) and for existing sub-elements as well (e.g. `error`) to build a good, consistent name. E.g. prefer the consistent use of `error.total` rather than introducing `failures`, `failed.total` or `errors`.` `
-* Avoid having sub-metrics under a name that is also a metric (e.g. do not create names like `es.repositories.elements`,` es.repositories.elements.utilization`; use` es.repositories.element.total` and` es.repositories.element.utilization `instead). Such metrics are hard to handle well in Elasticsearch, or in some internal structures (e.g. nested maps).
+The **hierarchy** of segments should be built by putting "more common" segments at the beginning. This facilitates the creation of new metrics under a common namespace.
+Each element in the metric name specializes or describes the prefix that precedes it. Rule of thumb: you could truncate the name at any segment, and what you're left with is something that makes sense by itself.
 
-Keep the hierarchy compact: do not add elements if you don’t need to. There is a description field when registering a metric, prefer using that as an explanation. \
-For example, if emitting existing metrics from node stats, do not use the whole “object path”, but choose the most significant terms.
+Example: Prefer `.docs.deleted.total` over `.deleted.docs.total` so `.docs.ingested.total` could be added later.
 
-The metric name can be generated but there should be no dynamic or variable content in the name: that content belongs to a **dimension** (attributes/labels).
+Recommendations:
+- When adding new metrics, [look for existing segments](#inspect-registered-metrics) as prefix in your domain. Also keep language consistent across the hierarchy, re-using segment names / terminology when possible.
+E.g. stick to using `.error.total` if already in use rather than introducing `.failures.total`.
```

Comment:
I understand these are supposed to be suffixes, but it's not clear. Could you add some explanation further up that these examples beginning with `.` would be added to some `es.whatever` prefix?